docs/intro/contributing.txt

```diff
@@ -103,26 +116,101 @@ Download the Django source code repository using the following command:
 
     $ git clone https://github.com/django/django.git
 
+Now that you have a local copy of Django, you can install it just like you would
+install any package using ``pip``. The most convenient way to do so is by using
+a *virtual environment* (or virtualenv) which is a feature built into Python
+that allows you to keep a separate directory of installed packages for each of
+your projects so that they don't interfere with each other.
+
+It's a good idea to keep all your virtualenvs in one place, for example in
+``.virtualenvs/`` in your home directory. Create it if it doesn't exist yet:
+
+.. code-block:: console
+
+    $ mkdir ~/.virtualenvs
+
+Now create a new virtualenv by running:
+
+.. code-block:: console
+
+    $ python3 -m venv ~/.virtualenvs/djangodev
+
+The path is where the new environment will be saved on your computer.
+
+If you are on Windows, run these commands in the Git Bash shell:
+
+.. code-block:: none
+
+    > mkdir C:Users/name/virtualenvs
+    > C:Python34/python -m venv C:Users/name/virtualenvs/djangodev
+
+Where ``Python34`` is the location of your Python installation, and ``name`` is
+your username.
+
 .. note::
 
-    For users who wish to use `virtualenv`__, you can use:
+    On some older versions of Python 3 the above command might fail. In that
+    case you can create the virtualenv through a Python package instead:
 
     .. code-block:: console
 
-        $ pip install -e /path/to/your/local/clone/django/
+        $ pip3 install virtualenv
+        $ virtualenv --python=`which python3` ~/.virtualenvs/djangodev
+
+    Or on Windows:
```

Comment:
As far as I know, this issue only affects Ubuntu (https://bugs.launchpad.net/ubuntu/+source/python3.4/+bug/1290847) so Windows/Mac aren't affected.  So "On older versions of Python 3" should be replaced with "On Ubuntu" I think.